### DISCUSSION TITLE
Welcome to conky Discussions!

### DISCUSSION THREAD

**Discussion by brndnmtthws** [brndnmtthws/conky#1208] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by johnmc98683** [comment]
I'm a long-time conky user/fan. Geriatric geek. Linux Mint user. I would like to ask questions and have found no reliable sources of answers besides the ubuntu forums, and those don't always satisfy.
johnmc

**Comment by rafimrfdn** [comment]
Sir, I run Conky on Hyprland. But it behaviour to always render conky on the top of all window. I need conky runs as background only. Can you please tell me how to config Conky on Hyprland?

**Reply by Caellian** [reply]
The default script with `out_to_wayland` set works fine. I added a Wayland section to the [wiki](https://github.com/brndnmtthws/conky/wiki/Window-Configuration#wayland) so check it out.

Note this thread is for introductions only, you should start a separate thread for your issue because this would've been considered spam if we were on an OG forum.

**Comment by ivy-reps** [comment]
1. @brndnmtthws, is there possibility to set milliseconds refresh interval? When I'm trying to use 0.5 seconds interval I get this:
![Screenshot from 2024-02-21 04-53-13](https://github.com/brndnmtthws/conky/assets/102535170/e312c028-675f-42a5-bb77-c18ebdd004a4)
I get that this is by design and that's why I'm writing here, not in issues. Also to many results in Issues search by `update interval` phrase. But anyway any chance to set milliseconds?
2. Also, is there simple way to make your default black theme just on whole desktop screen, not just on a part of a screen? Exactly, simple way, like wide screen argument or parameter?
Will be amazing to that black one on whole screen?
Or may be I can find User Config with your theme but wide screen? I saw User Config list, but haven't seen yours wide screened. I'm sorry if my question was asked already. I will understand if you ignore it.

**Reply by brndnmtthws** [reply]
The update interval issue looks like a bug, you should be able to resolve it by setting the value in conkyrc instead of passing it as a CLI argument.

You can set the window minimum width/height to match your desktop size, if you'd like that.

**Reply by brndnmtthws** [reply]
The CLI parsing bug is fixed in this: https://github.com/brndnmtthws/conky/pull/1728

**Reply by ivy-reps** [reply]
@brndnmtthws, thank you for fast reaction and hotfix! Amazing!
How much time it will take until it will be available through official Ubuntu apt repository?

**Comment by vntomilin88** [comment]
Hello community!
I have been an avid Conky user for years now, and since my last PC upgrade I ran into a problem. With a 32core processor displaying cpubars for each thread horizontally starts to look clunky. Is there any way to make a load bar display vertically instead of horizontally? What I imagine seeing is that with such option you would be able to create a processors load diagram similar to a wavelength diagram in an old school music player. I think such option would open quite a few new conky design ideas and make it even more visually appealing.

PS
I tried looking through the source code to find where the bar graphics are generated to perhaps add this function, but only got as far as special_node function.

**Reply by Caellian** [reply]
I recommend using cairo from lua for drawing these. See [displaying stuff](https://github.com/brndnmtthws/conky/wiki/Using-Lua-scripts-(Part-02)%3A-Displaying-stuff-in-conky) section in wiki - it allows you to go wild with customization though it's a bit of a time hog.

I agree that having vertical CPU bars would be cool, it would be a welcome addition.

**Comment by DjangoReinhard** [comment]
Hello community,
I'm a happy conky user too. I wrote my own theme with lua extensions and spent some time to get rid of existing hardware - to generate matching config files.
Actually I have an issue with sensors - I have various sensors of the same type, which all have various input files.
How can I distinguish the sensor and not the input file?
Actually I'm using exec, but read, that it wastes resources.
Then I tried to use lua function, instead of exec, but I wasn't successful in calling lua function from config texts.
The function exists (with conky_ prefix) but I get the error "attempt to call a nil value"

**Reply by Caellian** [reply]
Welcome!

> How can I distinguish the sensor and not the input file?

I don't understand what you meant with this question.

> Then I tried to use lua function, instead of exec, but I wasn't successful in calling lua function from config texts.
> [...] The function exists (with conky_ prefix) but I get the error "attempt to call a nil value"

Did you use `lua_load`?

There's a very detailed [wiki section](https://github.com/brndnmtthws/conky/wiki/How-to-Load-and-Use-Lua-Scripts) about this. I just cleaned it up and added some details with regards to function names and calling them from hooks - that people often find confusing so check it out. If you load multiple files use space separators until #2154 is merged (1/2 days), if the file name contains spaces you should replace them with `_` until #2154 is merged.

**Reply by DjangoReinhard** [reply]
Hi Caellian,

second first: yes of cause, I used lua_load. I build the config from within lua and I process texts like "${exec what ever}" with conky_parse - which will execute the script and return the result.
May be I got things wrong, but I thought, I could use ${lua what(ever)} to call a lua function named 'conky_what', but I get the error message from above (and yes - function definition is above the line where conky_parse will be executed)

first one: afaik the call to read a sensor is like "${hwmon nvme temp 1}"
So I can enumerate the temparatures, but not the sensor type (nvme). I have several sensors of type nvme, each one is connected to a different drive. So I'd like to know which of the sensors I'm reading 
(I read device/name to distinguish sensors, but don't know how to connect the name with the right value from conky.)

**Reply by Caellian** [reply]
> I could use ${lua what(ever)} to call a lua function named 'conky_what'

Yes, but, `conky.conf` is evaluated before `lua_load`. If you call `conky_parse` with `lua` variable from it, those functions will not had been loaded yet (`conky_parse` is eagerly evaluated), until after `conky.conf` is evaluated. `conky.text` is deferred until `lua_load` is done. Instead of `lua_load` you need to use `require("your source.lua")`; `conky.conf` parent directory is in Lua search paths. At that point `conky_parse(${lua function})` doesn't make sense anymore, so you can just replace it with a direct function call.

> don't know how to connect the name with the right value from conky

If you're already doing what I gathered you're doing in Lua, then use a lua loop to iterate over sensors and keep track of `i`th name:

```lua
sensors = {1: 0, 2: 0}

function conky_pre_hook()
    for sensor,value in pairs(sensors) do
        sensors[key] = conky_parse("${hwmon nvme temp 1}")
    end
end

function sensor_value(index)
  return sensors[index]
end
```

but you'll either need to manually draw the values using cairo, or use `conky.text` with `${lua sensor_value 1}` - I don't think `conky.text` can be dynamically updated.

**Comment by DjangoReinhard** [comment]
Thank you very much!
The loading order was very helpful so that I could solve my issues.
... and yes: it significantly reduced the system load. Great!

**Reply by Caellian** [reply]
Glad to hear it. I'll probably add loading order to wiki at some point.

**Comment by Barry-IA** [comment]
Found three potential errors, or at least gave me problems while running on a Raspberry Pi 4 using Trixie.
(1) In ‘Install’:

Original:
```
Install
$ mkdir -p ~/.conky
$ git clone https://github.com/woohgit/conkySimpleForecast.git ~/.conky/
$ cd ~/.conky/conkySimpleForecast
$ sudo make install
$ fc-cache -fv
```
Based on the ‘cd’ (#3) line it looks like the ‘git’ (#2) line should be going to ~/.conky/conkySimpleForecast  

(2) Appears there is a typographical error in run:
```
barry@raspberrypi:~ $ cat    ~/.conky/conkySimpleForecast/run
#!/bin/bash
# LC_ALL=en_US.UTF-8 conky -c conky_simpleforcast	# Spelling error!!!
LC_ALL=en_US.UTF-8 conky -c conky_simpleforecast
```

(3) In  ~/.conky/conkySimpleForecast/conky_simpleforecast   I needed to change
```
--	own_window_type = 'override',
	own_window_type = 'normal',
```
Using ‘override’ the programme seemed to run but without a visible display.  Changing to ‘normal’ I can see the window.

I still have a bit of work to do (mainly the units); hopefully this helps someone else with this nice-looking project.  (And I'll learn when someone comments on my comment!)  -- Barry
